anago

```diff
@@ -403,65 +403,18 @@ PROGSTEP[generate_release_notes]="GENERATE RELEASE NOTES"
 generate_release_notes () {
   local release_tars=$TREE_ROOT/_output-$RELEASE_VERSION_PRIME/release-tars
   local action="Update"
+  local nomock_flag=""
 
-  logecho -n "Generating release notes: "
-  logrun -s relnotes $RELEASE_VERSION_PRIME --release-tars=$release_tars \
-                     --branch=${PARENT_BRANCH:-$RELEASE_BRANCH} --htmlize-md \
-                     --markdown-file=$RELEASE_NOTES_MD \
-                     --html-file=$RELEASE_NOTES_HTML \
-                     --release-bucket=$RELEASE_BUCKET || return 1
-
-  logecho -n "Checkout master branch to make changes: "
-  logrun -s git checkout master || return 1
-
-  # The fetch and rebase before editing $CHANGELOG_FILE
-  # avoids merge conflicts if another release cut
-  # completed while we were building this one.
-  logecho -n "Fetch origin/master to get latest $CHANGELOG_FILE: "
-  logrun -s git fetch origin master || return 1
-  logecho -n "Rebase on origin/master before editing $CHANGELOG_FILE: "
-  logrun -s git rebase origin/master || return 1
-
-  if [[ ! -f $CHANGELOG_FILE ]]; then
-    cat<<EOF > $CHANGELOG_FILE
-<!-- BEGIN MUNGE: GENERATED_TOC -->
-
-<!-- END MUNGE: GENERATED_TOC -->
-
-<!-- NEW RELEASE NOTES ENTRY -->
-EOF
-    action="Add"
-    logecho -n "$CHANGELOG_FILE not found.  Creating: "
-    logrun -s git add $CHANGELOG_FILE
-  fi
-
-  logecho -n "Insert $RELEASE_VERSION_PRIME notes into $CHANGELOG_FILE: "
-  # Pipe to logrun() vs using directly, because quoting.
-  sed -i -e 's/<!-- NEW RELEASE NOTES ENTRY -->/&\n/' \
-         -e "/<!-- NEW RELEASE NOTES ENTRY -->/r $RELEASE_NOTES_MD" \
-   $CHANGELOG_FILE | logrun -s
-
-  logecho -n "Update $CHANGELOG_FILE TOC: "
-  logrun -s common::mdtoc $CHANGELOG_FILE || return 1
-
-  logecho -n "Committing $CHANGELOG_FILE: "
-  logrun -s git commit -am \
-            "$action $CHANGELOG_FILE for $RELEASE_VERSION_PRIME." \
-   || return 1
+  ((FLAGS_nomock)) && nomock_flag="--nomock"
 
-  # Sync $CHANGELOG_FILE to release-* branch and clear all others
-  if [[ "$RELEASE_BRANCH" =~ release- ]]; then
-    logecho -n "Checkout $RELEASE_BRANCH branch to make changes: "
-    logrun -s git checkout $RELEASE_BRANCH || return 1
-    logecho -n "Remove any previous CHANGELOG-*.md files: "
-    logrun -s git rm -f CHANGELOG-*.md || return 1
-    logecho -n "Copy master $CHANGELOG_FILE to $RELEASE_BRANCH branch: "
-    logrun -s git checkout master -- $CHANGELOG_FILE || return 1
-    logecho -n "Committing $CHANGELOG_FILE: "
-    logrun -s git commit -am \
-              "Add/Update $CHANGELOG_FILE for $RELEASE_VERSION_PRIME." \
-     || return 1
```

Comment:
Is this all (checking out the right things, committing, ...) by either:
- `anago`
- a GCB step
- ` k8s.io/release/cmd/release-notes/...`

🤔

You'll probably know that by heart, I will check on that to have 2 👓on it ;)

Reply:
Yes I think so. The only part where I was wondering if we can remove it was the `prepare_tree` `CHANGELOG-*.md` modifications made here:
https://github.com/kubernetes/release/blob/60d119b57b71c2691cc29d533dd1736493a45cbc/anago#L562-L574

:point_up: I think we do not need this any more, because I guess the intention was to cleanup the CHANGELOG files as early as possible after release branch creation. In our case we would cut the first beta after the release branch has been created, which would cleanup the CHANGELOG files in any case...